pkg/pillar/diskmetrics/usage.go

```diff
@@ -57,8 +69,21 @@ func SizeFromDir(log *base.LogObject, dirname string) (uint64, error) {
 				log.Tracef("Dir %s size %d\n", filename, size)
 				totalUsed += size
 			} else {
-				log.Tracef("File %s Size %d\n", filename, location.Size())
-				totalUsed += uint64(location.Size())
+				// FileInfo.Size() returns the provisioned size
+				// Sparse files will have a smaller allocated size than provisioned
+				// Use full syscall.Stat_t to get the allocated size
+				allocatedBytes, err := StatAllocatedBytes(filename)
+				if err != nil {
+					log.Errorf("StatAllocatedBytes: %s failed %s treating as fully allocated\n", filename, err)
+					log.Tracef("File %s Size %d\n", filename, location.Size())
```

Comment:
No need for two loglines I think. 
But would you ever see this?

I guess it could happen if the file is deleted between f.Readdir and this Stat, but is there some other case?
(And I don't know if location.Size() calls Stat internally or if it has that from the directory entry; if the former it could also return zero I assume.)

Reply:
Actually I'm glad you brought up these logs, the trace one should probably move down to the totalBytes+= line, I'll update this.  At least one other case stat could fail with ENOMEM no kernel memory.

Reply:
log lines addressed in latest push, moved trace back out one scope level